libs/wingc/src/lsp/mod.rs

```diff
@@ -0,0 +1,279 @@
+use crate::{
+	diagnostic::DiagnosticLevel,
+	parser::Parser,
+	type_check,
+	wasm_util::{combine_ptr_and_length, ptr_to_string},
+	Diagnostics,
+};
+use itertools::Itertools;
+use lsp_types::{
+	CompletionItem, CompletionResponse, Position, Range, SemanticToken, SemanticTokens, SemanticTokensResult, Url,
+};
+use std::{
+	cell::RefCell,
+	collections::{HashMap, HashSet},
+};
+use tree_sitter::Tree;
+
+use self::{completions::completions_from_ast, semantic_token::semantic_token_from_ast};
+
+mod completions;
+mod semantic_token;
+
+/// The result of running wingc on a file
+pub struct FileData {
+	/// Text data contained in the file (utf8)
+	pub contents: String,
```

Comment:
```suggestion
	/// Text data contained in the source file (utf8) for matching up with diagnostics
	/// TODO: how to support multiple source files
	pub contents: String,
```

Reply:
This struct is stored in a hashmap, mapping `file -> FileData`. So it currently supports "multiple files". Although what that may technically mean in the future could change. Either way not sure the TODO is needed.

Reply:
Oh ok, got it. I was just a little confused at first when I saw the struct was about the output of the compiler and it contained the input/source code of compilation - but in the context of the LSP I figured it made sense